packages/codegen/src/generateEntityCodegenFile.ts

```diff
@@ -594,7 +594,11 @@ function fieldHasDefaultValue(field: PrimitiveField | EnumField): boolean {
   );
 }
 
-function generatePolymorphicTypes(meta: EntityDbMetadata) {
+function generatePolymorphicTypes(meta: EntityDbMetadata, baseEntity?: EntityDbMetadata) {
+  // If this entity has a base class we will filter out any polymorphic fields that are from the base class
+  if (meta.baseClassName && baseEntity) {
+    meta.polymorphics = meta.polymorphics.filter((p) => !baseEntity.polymorphics.includes(p));
+  }
```

Comment:
@0ces I'm not sure this new code is really doing anything, because:

a) `meta.polymorphics` should not have any base-class fields in it, i.e. the code here:

```
    this.polymorphics = polymorphicRelations(config, table).map((rc) =>
      newPolymorphicField(config, table, this.entity, rc),
    );
```

When running for, say, AdminUser, is not going to find any poly fields from the User table.

b) When I comment this out locally, and rerun `yarn codegen`, there are no code changes.

My guess is that the primary bug fix is the `fields` -> `allFields` change in `PolymorphicReference.ts`, and this change in `generateEntityCodegenFile` is not necessary?

Reply:
You're right, this change isn't needed, just the `allFields` does it.